hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/BasicTransactionalTestCase.java

```diff
@@ -151,6 +154,123 @@ public Void call() throws Exception {
 	}
 
 	@Test
+	@TestForIssue( jiraKey = "BZ-1389846")
+	public void testAAAEntityAndManyToOneCachedWhenLoaded() throws Exception {
+		Statistics stats = sessionFactory().getStatistics();
+
+		stats.clear();
+
+		State state = new State();
+		state.setName( "Washington" );
+		County county = new County();
+		county.setName( "King" );
+		county.setState( state );
+
+		Transaction txn = null;
+		Session s = null;
+
+		beginTx();
+		try {
+			s = openSession();
+			txn = s.beginTransaction();
+			s.persist( state );
+			s.persist( county );
+			txn.commit();
+			s.close();
+		}
+		catch (Exception e) {
+			setRollbackOnlyTx( e );
+		}
+		finally {
+			commitOrRollbackTx();
+		}
+
+		stats.clear();
+
+/*
+		beginTx();
+		try {
+			s = openSession();
+			txn = s.beginTransaction();
+			// Load the County which has an eager many-to-one.
+			// This should load both the County and State entities from the 2lc.
+			county = (County) s.get( County.class, county.getId() );
+			assertTrue( Hibernate.isInitialized( county ) );
+			assertTrue( Hibernate.isInitialized( county.getState() ) );
+			txn.commit();
+			s.close();
+		}
+		catch (Exception e) {
+			setRollbackOnlyTx( e );
+		}
+		finally {
+			commitOrRollbackTx();
+		}
+
+		assertEquals( 1, stats.getSecondLevelCacheStatistics( County.class.getName() ).getHitCount() );
+		assertEquals( 1, stats.getSecondLevelCacheStatistics( State.class.getName() ).getHitCount() );
+*/
+		sessionFactory().getCache().evictEntityRegions();
+		stats.clear();
+
```

Comment:
The test passes if the following line is changed to:

`Thread.sleep( 60000 );`

This works because the `TransientCacheValue#maxIdle=60000` will expire with the longer sleep time.